hydromt/models/model.py

```diff
@@ -158,58 +188,12 @@ def api(cls) -> Dict:
                 _api.update(getattr(base_cls, "_API", {}))
         return _api
 
-    def _check_get_opt(self, opt):
-        """Check all opt keys and raise sensible error messages if unknown."""
-        for method in opt.keys():
-            m = method.strip("0123456789")
-            meth = self._get_sub_method(m)
-            if not callable(meth):
-                raise ValueError(f'Model {self._NAME} has no method "{method}"')
-        return opt
-
-    def _get_sub_method(self, method_name: str) -> Optional[Callable]:
-        # temporary code until the component logic is added.
-        # now we need this to be able to call region.create
-        obj = self
-        for subojb in method_name.split("."):
-            try:
-                obj = getattr(obj, subojb)
-            except AttributeError:
-                return None
-        func = cast(Callable, obj)
-        return func
-
-    def _run_log_method(self, method, *args, **kwargs):
-        """Log method parameters before running a method."""
-        method = method.strip("0123456789")
-        func = self._get_sub_method(method)
-        if not callable(func):
-            raise ValueError(f'Model {self._NAME} has no method "{method}"')
-        signature = inspect.signature(func)
-        # combine user and default options
-        params = {}
-        for i, (k, v) in enumerate(signature.parameters.items()):
-            if k in ["args", "kwargs"]:
-                if k == "args":
-                    params[k] = args[i:]
-                else:
-                    params.update(**kwargs)
-            else:
-                v = kwargs.get(k, v.default)
-                if len(args) > i:
-                    v = args[i]
-                params[k] = v
-        # log options
-        for k, v in params.items():
-            if v is not inspect._empty:
-                self.logger.info(f"{method}.{k}: {v}")
-        return func(*args, **kwargs)
-
     def build(
         self,
-        region: Optional[dict] = None,
+        *,
```

Comment:
The `*` means that all arguments after are keyword arguments. You cannot simply call `model.build({"bbox": [1,2,3,4]}, "some_stuff", "and_more")`. You need to specify `model.build(region={"bbox": [1,2,3,4]})`. I prefer this way when there are 3 or more arguments, as you otherwise cannot see which parameter means what.